module/zfs/arc.c

```diff
@@ -5137,8 +5161,10 @@ arc_is_overflowing(void)
 	 * in the ARC. In practice, that's in the tens of MB, which is low
 	 * enough to be safe.
 	 */
-	return (aggsum_lower_bound(&arc_sums.arcstat_size) >=
-	    (int64_t)arc_c + overflow);
+	int64_t over = aggsum_lower_bound(&arc_sums.arcstat_size) -
+	    arc_c - overflow / 2;
+	return (over < 0 ? ARC_OVF_NONE :
+	    over < overflow ? ARC_OVF_SOME : ARC_OVF_SEVERE);
```

Comment:
`over < overflow` is off by `overflow / 2` from the previous logic here, so `SEVERE` is not reached at the same point we were overflowing before.
```
a >= c + v
vs
a - c - v/2 >= v
```
It's not clear to me if that is intentional, but the tunable documentation in `zfs(4)` should be updated to reflect the new behavior either way.

Reply:
It is intentional, since now there are two thresholds instead of one before.  I'll update the man page.

Reply:
Updated.